### PR TITLE
fix codeql and setup ossf scorecard workflow

### PR DESCRIPTION
**Pull request by mmorel-35**
## Which problem is this PR solving?
- fix pin dependencies vulnerabilities from ossf and codeql

## Description of the changes
- fix codeql and setup ossf scorecard workflow

## How was this change tested?
-  CI

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: tests/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # The Dockerfile's resulting image is purpose-built for executing Jaeger Operator e2e tests within the OpenShift release (https://github.com/openshift/release) using Prow CI.
-FROM golang:1.21
```
Comment (iblancasa) [inline tests/Dockerfile:2]: Why is this change needed?

File: Dockerfile.asserts [hunk 2]
```diff
@@ -27,7 +27,7 @@ RUN GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -o ./index -a ./tests/assert-
 
 # Use the curl container image to ensure we have curl installed. Also, it is a
 # minimal container image
-FROM curlimages/curl:7.81.0
+FROM curlimages/curl@sha256:4bfa3e2c0164fb103fb9bfd4dc956facce32b6c5d47cc09fcec883ce9535d5ac
```
Comment (iblancasa) [inline Dockerfile.asserts:30]: Why is this change needed?

File: tests/Dockerfile (hunk 1 above)
Comment (mmorel-35) [reply tests/Dockerfile:2]: This is an OSSF recommandation. Pining version avoid certain vulnerabilities.

**Review by iblancasa — APPROVED** [review]
